RA- 06-create-classes-for-entities

DenisGubernatorov opened the pull request:
Создание классов для сущностей согласно ТЗ, добавление изображений, файл с объектом-заглушкой возвращающим список категорий

DenisGubernatorov committed:
RA-06-create-classes-for-entities
1) Доработка логики загрузки фрагментов. При загрузке стартового экрана он не добавляется в backstack
2)  Statusbar и navigationbar сделан контрастными

DenisGubernatorov committed:
RA-06-create-classes-for-entities
1) Создание дата-классов для сущностей  Category, Recipe, Ingredient

DenisGubernatorov committed:
RA-06-create-classes-for-entities
1) Добавление изображений в assets

DenisGubernatorov committed:
RA-06-create-classes-for-entities
1) Создание файла recipes.kt с синглтон объектом STUB, возвращающим список категорий.

aGroupoid commented on app/src/main/java/com/example/recipesapp/MainActivity.kt:
```diff
@@ -18,23 +18,25 @@ class MainActivity : AppCompatActivity() {
     }
 
-    private inline fun <reified T : Fragment> load() {
+    private inline fun <reified T : Fragment> loadFragment(addToBackStack: Boolean) {
```
В Kotlin можно сделать в функции значение по-умолчанию loadFragment(addToBackStack: Boolean = true), тогда в общем случае нет необходимости закидывать true в функцию, только false при необходимости

aGroupoid commented on app/src/main/java/com/example/recipesapp/Ingredient.kt:
```diff
@@ -0,0 +1,7 @@
+package com.example.recipesapp
+
+data class Ingredient(
+    val quantity: Double,
```
Посмотри еще раз внимательнее на формат. Строки оборачиваются в двойные кавычки, а здесь числовое значение

aGroupoid commented on app/src/main/java/com/example/recipesapp/Recipe.kt:
```diff
@@ -0,0 +1,7 @@
+data class Recipe(
+    val id: Int,
+    val title: String,
+    val ingredients: List<Ingredient>,
```
Пролистай вниз файла, там еще есть параметры. Добавь их тоже
https://gist.github.com/ievetrov/02643bec17c40632a74014fb4281da89

aGroupoid commented on app/src/main/java/com/example/recipesapp/MainActivity.kt:
```diff
@@ -18,23 +18,25 @@ class MainActivity : AppCompatActivity() {
             replace<T>(R.id.mainContainer)
-            addToBackStack(null)
+            if (addToBackStack) {
+                addToBackStack(null)
```
немного странно выглядит такой нейминг переменной. Давай переименуем isInBackstack или как-то так

DenisGubernatorov committed:
RA-06-create-classes-for-entities
1) Ingredient - изменения типа для quantity
2) Recipe добавлены пропущенные свойства
3) loadFragment изменено имя флага, что нужно добавлять в backStack. Сделан true по умолчанию

aGroupoid reviewed (approved):
Молодец, всё верно!
